src/DataTable/TableFooter.jsx

```diff
@@ -1,15 +1,29 @@
 /* eslint-disable react/require-default-props */
 import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
 import RowStatus from './RowStatus';
 import TablePagination from './TablePagination';
 
-const TableFooter = () => (
-  <div className="pgn__data-table-footer">
-    <RowStatus
-      className="pgn__data-table-footer-row-status"
+const TableFooter = ({ className, LeftComponent, RightComponent }) => (
+  <div className={classNames(className, 'pgn__data-table-footer')}>
+    <LeftComponent
+      className="pgn__data-table-footer-left"
     />
-    <TablePagination />
+    <RightComponent />
```

Comment:
I think we should prefer children here something like:

```
const TableFooter = ({ children }) => (
  <div className="...png__data-table-footer">
    {children}
  </div>
);

TableFooter.defaultProps = {
  children: (
    <>
      <TablePagination />
      <RowStatus />
    </>
  )
}

Reply:
OK so. 
Pros: Doing this makes it infinitely flexible. That's nice. It's also in line with how the TableData component is written.
Cons: I think part of having a design system is that you're pushing people to follow good design patterns, and I can Imagine people doing all sorts of things. Also, they can just write their own custom footer at that point?

Reply:
My take:

Make it easy to do the right thing. Make it possible to do something with didn't think of, and trust our team to use this power judiciously. If our components do the right things by default, escape hatches should be infinitely flexible via composability or render props. We should be more concerned with building unnecessarily limiting constraints in our components than with abuse of the flexibility we provide.

- By default or without configuration components should do the common, desirable thing
- Desirable variants of a pattern should be easy to create without heavy customization (not related to this particular scenario)
- Escape hatches should provide as much flexibility as possible.